circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/annotation/HiveTableAnnotator.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (C) 2016-2019 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.core.annotation;
+
+import java.util.Map;
+
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.thrift.TException;
+
+import com.google.common.base.Supplier;
+
+import com.hotels.bdp.circustrain.api.CircusTrainException;
+import com.hotels.hcommon.hive.metastore.client.api.CloseableMetaStoreClient;
+
+public class HiveTableAnnotator {
+
+  private Supplier<CloseableMetaStoreClient> replicaMetaStoreClientSupplier;
+
+  public HiveTableAnnotator(Supplier<CloseableMetaStoreClient> replicaMetaStoreClientSupplier) {
+    this.replicaMetaStoreClientSupplier = replicaMetaStoreClientSupplier;
+  }
+
+  public void annotateTable(String databaseName, String tableName, Map<String, String> parameters)
+    throws CircusTrainException {
+    if (parameters.isEmpty()) {
+      return;
+    }
+    try (CloseableMetaStoreClient client = replicaMetaStoreClientSupplier.get()) {
+      Table table = client.getTable(databaseName, tableName);
+      Map<String, String> tableParameters = table.getParameters();
```

Comment:
Maybe: `new LinkedHashMap(table.getParameters())` Just in case the returned map is immutable. Linked to keep insertion order, not sure if that matters much but won't hurt.

Reply:
Yep that makes sense, updated